### PR TITLE
feat(Interactions): context menu items

### PR DESCRIPTION
**Pull request by advaith1**
... contextybois? rightclickybois?

* https://github.com/discord/discord-api-docs/pull/3614

TODO: update like all of the interaction docs links lol

### CODE REVIEW

**Comment by vladfrangu** [comment]
This needs yet another rebase 👀

File: payloads/v9/_interactions/applicationCommands.ts [hunk 1]
```diff
@@ -41,6 +45,12 @@ export interface APIApplicationCommand {
 	default_permission?: boolean;
 }
 
+enum ApplicationCommandType {
```
Comment (vladfrangu) [inline payloads/v9/_interactions/applicationCommands.ts:48]:
```suggestion
export const enum ApplicationCommandType {
```

File: payloads/v9/_interactions/applicationCommands.ts [hunk 2]
```diff
@@ -267,16 +310,44 @@ export const enum ApplicationCommandPermissionType {
...
+export type APIMessageApplicationCommandGuildInteraction =
+	APIGuildInteractionWrapper<APIMessageApplicationCommandInteraction>;
+
+export type APIContextMenuInteraction = APIApplicationCommandInteractionWrapper<APIContextMenuInteractionData>;
+
+export type APIContextMenuDMInteraction = APIDMInteractionWrapper<APIContextMenuInteraction>;
+
+export type APIContextMenuGuildInteraction = APIGuildInteractionWrapper<APIContextMenuInteraction>;
+
+export type APIApplicationCommandInteraction =
```
Comment (vladfrangu) [inline payloads/v9/_interactions/applicationCommands.ts:349]: shouldn't this be a union of the previous ones (the context menu and chat input ones)?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: rest/v9/interactions.ts [hunk 3]
```diff
@@ -27,7 +27,8 @@ export type RESTGetAPIApplicationCommandResult = APIApplicationCommand;
 /**
  * https://discord.com/developers/docs/interactions/slash-commands#create-global-application-command
  */
-export type RESTPostAPIApplicationCommandsJSONBody = Omit<APIApplicationCommand, 'id' | 'application_id'>;
+export type RESTPostAPIApplicationCommandsJSONBody = Omit<APIApplicationCommand, 'id' | 'application_id' | 'type'> &
```
Comment (vladfrangu) [inline rest/v9/interactions.ts:30]: This doesn't realize description is not needed for 2 of the 3 types

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by didinele — APPROVED** [review]
Other than the attachment below, LGTM

**Review by didinele — APPROVED** [review]

File: payloads/v9/_interactions/_applicationCommands/contextMenu.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import type { APIUser } from '../../index';
+import type { Snowflake } from '../../../../globals';
+import type { APIMessage } from '../../channel';
+import type { APIDMInteractionWrapper, APIGuildInteractionWrapper } from '../base';
+import type {
+	APIApplicationCommandInteractionWrapper,
+	APIBaseApplicationCommandInteractionData,
```
Comment (vladfrangu) [inline payloads/v9/_interactions/_applicationCommands/contextMenu.ts:7]: Uuu... don't think we should have this exported

File: payloads/v9/_interactions/_applicationCommands/contextMenu.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
+import type { APIUser } from '../../index';
```
Comment (vladfrangu) [inline payloads/v9/_interactions/_applicationCommands/contextMenu.ts:1]: Point to users

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**